dm/worker/subtask.go

```diff
@@ -247,70 +250,70 @@ func (st *SubTask) callCurrCancel() {
 	st.RUnlock()
 }
 
-// fetchResult fetches units process result
-// when dm-unit report an error, we need to re-Process the sub task.
-func (st *SubTask) fetchResult(pr chan pb.ProcessResult) {
-	defer st.wg.Done()
+// fetchResultAndUpdateStage fetches process result, call Pause of current unit if needed and updates the stage of subtask.
+func (st *SubTask) fetchResultAndUpdateStage(pr chan pb.ProcessResult) {
+	defer st.resultWg.Done()
 
-	select {
-	case <-st.ctx.Done():
-		// should not use st.currCtx, because will do st.currCancel when Pause task,
-		// and this function will return, and the unit's Process maybe still running.
-		return
-	case result := <-pr:
-		// filter the context canceled error
-		errs := make([]*pb.ProcessError, 0, 2)
-		for _, err := range result.Errors {
-			if !unit.IsCtxCanceledProcessErr(err) {
-				errs = append(errs, err)
-			}
+	result := <-pr
```

Comment:
Not sure if this receive can cause dead lock or other severe side effect it ignores all levels' ctx.Done event

Reply:
It will. I added a comment in interface, maybe we should add unit tests to ensure this 🤔 

https://github.com/pingcap/dm/pull/1994/files#diff-a4af489b56eb811534ce618e8b923fb135156cfec5f40bb565e03d3b490d3ddcR44